### PR TITLE
feat(gui): configurable output file name

### LINKED ISSUE #452: Configurable output file
The output file is not configurable. This becomes a problem when you output a file, open it up in a DAW (e.g. FL Studio), and want to bounce out a new converted voice file, causing a crash while writing because the file handle is still held by the DAW.

Recommended features:
- A new output text box that allows specifying the output file
- The default output filename should follow the same `.out` pattern as before, but have an automatically incrementing index if the output file already exists.

I've already implemented this and will be submitting a pull request shortly.

### PR DESCRIPTION
**Pull request by alexanderkoumis**
Closes #452

Implements:
- Output file box
- Auto incrementing default output path

Screenshot 2023-04-22 170051

### CODE REVIEW

File: src/so_vits_svc_fork/gui.py [hunk 1]
```diff
@@ -255,10 +268,19 @@ def main():
...
                 sg.Button("Play", key="play_input"),
             ],
+            [
+                sg.Text("Output audio path"),
+                sg.Push(),
+                sg.InputText(key="output_path"),
+                sg.FileSaveAs(
+                    initial_folder=".",
+                    key="output_path_browse",
+                    file_types=(("WAV", ".wav"), ("MP3", ".mp3"))),
```
Comment (34j) [inline src/so_vits_svc_fork/gui.py:282]: I think more file types are supported.

**Review by 34j — CHANGES_REQUESTED** [review]

**Comment by 34j** [comment]
Thanks for your contrib.

File: src/so_vits_svc_fork/gui.py (hunk 1 above)
Comment (alexanderkoumis) [reply src/so_vits_svc_fork/gui.py]: The file save output will now show all supported audio encoding types:

Screenshot 2023-04-22 163506

At inference time the output extension will also be checked to ensure it is supported:

Screenshot 2023-04-22 164421

File: src/so_vits_svc_fork/gui.py [hunk 2]
```diff
@@ -57,6 +58,54 @@ def delete_preset(name: str) -> dict:
...
+
+def validate_output_file_type(output_path: Path) -> bool:
+    supported_file_types = sorted(
+        [f".{extension.lower()}" for extension in sf.available_formats().keys()]
+    )
+    if not output_path.suffix:
+        sg.popup_ok(
+            "Error: Output path missing file type extension, enter "
+            + "one of the following manually:"
+            + os.linesep
```
Comment (34j) [inline src/so_vits_svc_fork/gui.py:92]: I think using /n is better if it works, whatever

**Review by 34j — CHANGES_REQUESTED** [review]

File: src/so_vits_svc_fork/gui.py (hunk 2 above)
Comment (alexanderkoumis) [reply src/so_vits_svc_fork/gui.py]: I thought Windows would need `\r\n` but I just tried and it works with just `\n` as well, so I've now switched over.

**Review by 34j — APPROVED** [review]

**Comment by 34j** [comment]
@allcontributors add alexanderkoumis code

**Comment by 34j** [comment]
@alexanderkoumis
It seems that some people find this change not very convenient. If you don't mind, can you elaborate it further? (#492)

**Pull request merged**